xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-api/src/main/java/org/xwiki/panels/internal/script/OrphanedPagesItem.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.panels.internal.script;
+
+import java.util.List;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.xwiki.text.XWikiToStringBuilder;
+
+/**
+ * Contains the information required to display the orphaned page tree, namely a list of orphaned page references, the
+ * next offset, and if their is more orphan pages to be queried.
+ *
+ * @version $Id$
+ * @since 13.1RC1
+ * @since 12.10.4
+ * @since 12.6.8
+ */
+public class OrphanedPagesItem
```

Comment:
The "Item" suffix doesn't sound good to me. It makes me think it represents a single orphan page. Maybe ``OrphanPagesResults``. Now, the role served by this class is generic. I know @tmortagne created something similar for the Extension Manager ([``IterableResult``](https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-api/src/main/java/org/xwiki/extension/repository/result/IterableResult.java)) so I'm wondering if we shouldn't have this in a common place.